cluster.tf

```diff
@@ -32,10 +32,18 @@ resource "aws_eks_cluster" "this" {
     aws_iam_role_policy_attachment.cluster_AmazonEKSServicePolicy,
     aws_cloudwatch_log_group.this
   ]
+}
+
+resource "null_resource" "wait_for_cluster" {
+  depends_on = [
+    aws_eks_cluster.this[0]
+  ]
+
   provisioner "local-exec" {
-    command = <<EOT
-    until curl -k -s ${aws_eks_cluster.this[0].endpoint}/healthz >/dev/null; do sleep 4; done
-  EOT
+    command = var.manage_aws_auth ? var.wait_for_cluster_cmd : ""
```

Comment:
@max-rocket-internet this would optionally perform a no-op if manage_aws_auth is disabled? I may have to use count, but was unsure on wiring dependencies on the kubernetes_config_map resource.

Reply:
Like it but have you tested this in both situations? No errors or issues with having a command of `""`

Reply:
I get this error if `manage_aws_auth = false`:

```
Error: local-exec provisioner command must be a non-empty string
```

So I think you need to do this:

```hcl
resource "null_resource" "wait_for_cluster" {
  count      = var.manage_aws_auth ? 1 : 0

  depends_on = [
    aws_eks_cluster.this[0]
  ]

  provisioner "local-exec" {
    command = var.wait_for_cluster_cmd
    environment = {
      ENDPOINT = aws_eks_cluster.this[0].endpoint
    }
  }
}
```

Reply:
This is what I wanted to in my initial attempt, but I was not sure how this would be included as a dependency in the aws_auth module. I will try this and test without aws_auth. Thanks again.